webapp/src/components/properties/user/user.tsx

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect, useState} from 'react'
+import Select from 'react-select'
+
+import {IUser} from '../../../user'
+import octoClient from '../../../octoClient'
+
+import './user.scss'
+import {getSelectBaseStyle} from '../../../theme'
+
+type Props = {
+    value: string,
+    readonly: boolean,
+    onChange: (value: string) => void,
+}
+
+const UserProperty = (props: Props): JSX.Element => {
+    const [users, setState] = useState<Array<IUser>>([])
+
+    useEffect(() => {
```

Comment:
This is going to create a getWorkspaceUsers call for each UserProperty instance rendered, I think makes a lot of sense to have the list of users of the workspace in something like context manager. Whenever you switch the workspace, you get the users of the workspace, and wherever you need to show the list of users or get an specific user, you can use it from the context.

Reply:
Thanks a lot for that catch. Didn't knew about React Contexts. I've refactored this now.